gradle-plugin/build.gradle.kts

```diff
@@ -131,16 +131,19 @@ tasks.named("processTestResources").configure {
     dependsOn(writeTestPropsTask)
 }
 
+val cleanupTemporaryTestDir = tasks.register("cleanupTemporaryTestDir", Delete::class.java) {
+    delete = setOf(tempTestDir)
+}
+
 tasks.test {
+    dependsOn(cleanupTemporaryTestDir)
     dependsOn(":api:publishAllPublicationsToTestRepository")
     dependsOn(":gradle-plugin:publishAllPublicationsToTestRepository")
     dependsOn(":symbol-processing:publishAllPublicationsToTestRepository")
 
     jvmArgumentProviders.add(RelativizingLocalPathProvider("java.io.tmpdir", tempTestDir))
 
-    doFirst {
-        tempTestDir.deleteRecursively()
-    }
+    maxParallelForks = gradle.startParameter.maxWorkerCount / 2
```

Comment:
how did you determine that `maxWorkerCount / 2` is a good value? Does it work better than a value of just `2`?

Reply:
Thanks for testing this. Good to know that you tested on GHA mac runners. What about local builds? What is best for those?

I think there is no way to get a value that works in all configurations. I strongly discourage any changes to these values that is based on the `maxWorkerCount` since this can be quite a high value on some machines.